src/healthcheck/HealthChecker.ts

```diff
@@ -286,7 +286,11 @@ class Plugin {
         })
         .catch((error) => {
           this.status = failure;
-          this.statusReason = error.message;
+          if (error instanceof Error) {
+            this.statusReason = error.message;
```

Comment:
if `.message` is a property accessor, and it throws on access, what will happen here? I think there might need to be another .catch to catch clause that catches that case (paranoid, but we've seen this in real code, and instrumentation like this should never fail)

ditto for below.... I assume `.statusReason` will eventually have `.toString()` conversion done to it, and if that throws, would the app blow up?